src/Core/Containers/VectorArray.hpp

```diff
@@ -3,20 +3,33 @@
 #include <Core/Containers/AlignedStdVector.hpp>
 #include <Core/RaCore.hpp>
 
+#include <type_traits> // std::is_integral
+
 namespace Ra {
 namespace Core {
 
+/// Internal class used to compute types and values used by VectorArray, to handle both
+/// eigen matrices and integral types
+template <typename V, bool isIntegral>
+struct VectorArrayTypeHelper {};
+
 /// This class is a wrapper around a std::vector of Core::Vectors.
 /// which allow to use the stdlib's dynamic array implementation, yet pass it as
 /// a matrix when Eigen needs it with the getMap() method.
 template <typename V>
 class VectorArray : public AlignedStdVector<V>
 {
+  private:
+    using TypeHelper =
+        VectorArrayTypeHelper<V, std::is_integral<V>::value || std::is_floating_point<V>::value>;
```

Comment:
```suggestion
        VectorArrayTypeHelper<V, is_arithmetic_v<T>>;
```
Well, I understand your restriction to arithmetic types (integral and floating point) so that this class is compatible with Eigen functionalities that do not work if we store pointers in this array. Am I right ?

But, in this case, why not changing also the class name (its VectorArray and is not representative of scalar array if V is just an arithmetic  type)  and documentation as it is no more a "std::vector of Core::Vectors" as  it can store scalars (any integral type or floating point type) as well  ? 
Why can't we have any scalar type (using the proposed std::is_scalar) as well as Eigen Vectors ? Some eigen functionalities could be disabled for non eigen compatible type as you did for the alias scalar_type.

Reply:
We can replace `is_arithmetic_v` by `is_scalar`.
Regarding the name: `VectorArray` was already supporting `float` and `double`: this PR only add other types, nothing else. 

We might to want to change the name of this class and refactor its use in `Attrib`, however I believe this is out of the scope of this class.